src/Sema.zig

```diff
@@ -129,15 +135,29 @@ pub fn analyzeBody(
 ) CompileError!Zir.Inst.Index {
     // No tracy calls here, to avoid interfering with the tail call mechanism.
 
+    const parent_capture_scope = block.wip_capture_scope;
+
+    var wip_captures = WipCaptureScope{
+        .finalized = true,
+        .scope = parent_capture_scope,
+        .perm_arena = sema.perm_arena,
+        .gpa = sema.gpa,
+    };
+    defer if (wip_captures.scope != parent_capture_scope) {
```

Comment:
RIP my poor, sweet main loop :sob: 

ok for now; I'll come back to this later and make sure the machine code here ends up looking ok. there are some tricks we can do.